File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set = new HashSet<>();
        int l = 0;
        int longest = 0;
        int count = 0;

        for(int r = 0; r < s.length(); r++){
            while(set.contains(s.charAt(r))){
               System.out.println("REMOVED: " + s.charAt(l) + " L: " + l);
               set.remove(s.charAt(l)); 
               l++; 
               count = r-l; 
            }
            count++; 
            set.add(s.charAt(r));
            System.out.println("ADDED: " + s.charAt(r));
            longest = Math.max(longest, count);
        }
        return longest; 

        
    }
}

File: Data Structures & Algorithms/is-anagram/submission-1.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) return false;

        var matchedRes = new List<int>();


        foreach(char c in s){
            if(matchedRes.Contains(c)){
                break;
            }
            foreach(char j in t){
                if (c == j){
                    matchedRes.Add(c);
                }
            }
        }
        foreach (char c in matchedRes){
            Console.WriteLine(c);
        }


        return matchedRes.Count == s.Length;


    }
}
